modules/java/jar/build.xml.in

```diff
@@ -29,14 +29,33 @@
   </target>
 
   <target name="javadoc">
+    <copy file="@OpenCV_SOURCE_DIR@/doc/mymath.js"
+      todir="@OpenCV_BINARY_DIR@/doc/doxygen/html/javadoc" />
+
     <javadoc
       packagenames="org.opencv.*"
       sourcepath="java"
       destdir="@OpenCV_BINARY_DIR@/doc/doxygen/html/javadoc"
       Windowtitle="OpenCV @OPENCV_VERSION_PLAIN@ Java documentation"
       Doctitle="OpenCV Java documentation (@OPENCV_VERSION@)"
       bottom="Generated on ${timestamp} / OpenCV @OPENCV_VCSVERSION@"
-      />
+      failonerror="true"
+      encoding="UTF-8" charset="UTF-8" docencoding="UTF-8"
+      link="https://docs.oracle.com/javase/6/docs/api/"
+      additionalparam="--allow-script-in-comments"
+      >
+      <Header>
+         <![CDATA[
+            <script>
+              var url = window.location.href;
+              url = url.substring(0, url.lastIndexOf('/', url.indexOf('.html'))) + '/../../../mymath.js';
+              var script = document.createElement('script');
+              script.src = 'https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML,' + url;
```

Comment:
> /latest/

is not a good idea in general.

Please use `@OPENCV_MATHJAX_RELPATH@` here.

Reply:
it seems the variable is not replaced.

Reply:
Right!
This variable is defined in `doc/CMakeLists.txt` which is processed after java module.

Please move this variable definition into the main [CMakeLists.txt](https://github.com/opencv/opencv/blob/3.4.6/CMakeLists.txt#L499).